recipe_scrapers/keukenliefdenl.py

```diff
@@ -0,0 +1,57 @@
+# mypy: allow-untyped-defs
+
+from ._abstract import AbstractScraper
+from ._utils import get_minutes, get_yields, normalize_string
+
+
+class KeukenLiefdeNL(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "keukenliefde.nl"
+
+    def author(self):
+        return self.soup.find("meta", {"name": "author"}).get("content")
+
+    def title(self):
+        return self.soup.find("meta", {"property": "og:title"}).get("content")
+
+    def category(self):
+        return self.soup.find(
+            "div", {"class": "article-meta-item sp gerecht"}
+        ).getText()
+
+    def total_time(self):
+        return get_minutes(
+            self.soup.find("div", {"class": "article-meta-item sp tijd"}).get_text()
+        )
+
+    def yields(self):
+        return get_yields(
+            self.soup.find("div", {"class": "article-meta-item sp aantal"}).get_text()
+        )
+
+    def image(self):
+        return self.schema.image()
+
+    def ingredients(self):
+        ingredients = self.soup.find(
+            "div", {"id": "clipboard-ingredients"}
+        ).findChildren("li")
+
+        return [normalize_string(i.get_text()) for i in ingredients]
+
+    def instructions(self):
+        # Old recipes are written in paragraphs, new ones are a list.
+        paragraphs = self.soup.find("div", {"class": "preparation"}).find_all("p")
+        listitems = self.soup.find("div", {"class": "preparation"}).find_all("li")
+
+        instructions = [
+            normalize_string(item.get_text())
+            for item in paragraphs + listitems
+            if item.get_text()
+        ]
```

Comment:
I would suggest a small adjustment here:

```suggestion
        instructions = [
            normalize_string(item.get_text())
            for item in paragraphs + listitems
            if normalize_string(item.get_text())
        ]
```

This is to reduce the chance that any content that becomes an empty string after normalization (removal of whitespace, non-printable characters, and so on) can sneak through.

Python considers any non-empty string to be truthy (`True` when used in a boolean context like the `if` clause here) and so there's a chance that elements containing spaces (for example, a newline between an opening `p` tag and the closing one on a subsequent line) would pass the `if item.get_text()` check, and yet appear in the `instructions()` return value as an empty string.

I'm not sure whether that's a great explanation; let me know if I can help to clarify that.